lib/environments/meteorEnvironment.js

```diff
@@ -321,13 +321,7 @@ export default {
     ['.html', '.css'].forEach((ext: string) => {
       const moduleSpecifier = `${basePath}${ext}`;
       if (fs.existsSync(path.join(config.workingDirectory, moduleSpecifier))) {
-        if (config.get('useRelativePaths')) {
-          moduleSpecifiers.push('.'.concat(path.sep, path.basename(moduleSpecifier)));
-        } else {
-          // Strip the leading '.' off of the moduleSpecifier to turn it into a Meteor compliant
-          // absolute path.
-          moduleSpecifiers.push(moduleSpecifier.slice(1));
```

Comment:
To "turn off" `useRelativePaths` with this, you basically need to return a formatted version of `pathToModule`. Here's a simplified example:

```js
moduleNameFormatter({ pathToImportedModule }) {
  return pathToImportedModule.slice(2); // remove leading `./`
},
```

I think the array approach might work, but I don't like the special casing of one config value. I think it would be better if we can find a solution that will work for any configuration value. I have some thoughts about this.

It looks like we currently execute configurations starting with the user config, followed by the environment config, and ending with the base config. At each level of this configuration, we make a decision about whether we should merge the result or just return the value. This special cases some configuration options in potentially confusing ways. Also, it isn't as flexible as it could be. What if an environment configuration wanted to override some of the base config's values instead of simply being merged in? Or what if the user config wanted to do the same?

What if we reversed the order that configurations are executed (base, env, then user) and always pass down the previous value (similar to what we have done for `moduleName` and `moduleNameFormatter`, but more general, so the object would look more like `{ pathToImportedModule, pathToCurrentFile, value }`). This would allow environment and user configs to merge in previous values if they want, or to override them, which allows us to remove the special casing `MERGABLE_CONFIG_OPTIONS`.

Once this is done, we could choose to implement convenience configurations like `stripFileExtensions` in the base config's `moduleNameFormatter`. 

I think we should also remove the JSON configuration and require that every config value be a function, to simplify the logic in our `Configuration` class, but I don't think that is necessary for this.